src/main/java/edu/kit/compiler/semantic/NamespaceGatheringVisitor.java

```diff
@@ -134,44 +134,44 @@ public Void visit(ClassNode classNode) {
 
         @Override
         public Void visit(StaticMethodNode method) {
-            visitMethodNode(method);
-
-            if (hasMethod(method.getName())) {
-                semanticError(method, "method %s is already defined in class %s",
-                    stringTable.retrieve(method.getName()),
-                    stringTable.retrieve(classNode.getName()));
-            } else if (mainMethod.isPresent()) {
-                semanticError(method, "only a single static method is allowed in the entire program");
-            } else if (!stringTable.retrieve(method.getName()).equals("main")) {
-                semanticError(method, "only a single static method, which must "
+            this.visitMethodNode(method);
+
+            if (this.hasMethod(method.getName())) {
+                NamespaceGatheringVisitor.this.semanticError(method, "method %s is already defined in class %s",
```

Comment:
Could you please remove all the inserted `this` and formatting changes? They make it quite difficult to see the actual changes.

Reply:
Done